Submitted By: William Immendorf <will.immendorf@gmail.com>
Date: 2010-04-03
Initial Package Version: 0.6.25
Upstream Status: Have yet to submit it.
Origin: Self, with some fixes by BLFS Trac user Ringis.
Description: Various fixes for the Avahi initscripts. Run autoreconf -i -v after applying this patch.

diff -Naur avahi-0.6.25.orig/initscript/lfs/avahi avahi-0.6.25/initscript/lfs/avahi
--- avahi-0.6.25.orig/initscript/lfs/avahi	1969-12-31 18:00:00.000000000 -0600
+++ avahi-0.6.25/initscript/lfs/avahi	2010-04-03 11:30:50.040927032 -0500
@@ -0,0 +1,60 @@
+#!/bin/sh
+########################################################################
+# Begin $rc_base/init.d/avahi
+#
+# Description : Avahi daemon loader
+#
+# Authors     : William Immendorf - will.immendorf@gmail.com, with some fixes by BLFS Trac user Ringlis.
+#
+# Version     : 00.00
+#
+# Notes       : Based off of the LFS 6.4 sysklogd script. This controlls all but the AutoIP daemon (I haven't been able to start that).
+#
+########################################################################
+
+. /etc/sysconfig/rc
+. ${rc_functions}
+
+case "${1}" in
+	start)
+		boot_mesg "Starting the Avahi daemon..."
+		loadproc avahi-daemon -D
+
+		boot_mesg "Starting the Avahi mDNS/DNS-SD Server Conf. Daemon..."
+		loadproc avahi-dnsconfd -D
+		;;
+
+	stop)
+		boot_mesg "Stopping the Avahi mDNS/DNS-SD Server Conf. Daemon..."
+		avahi-dnsconfd -k
+		evaluate_retval
+
+		boot_mesg "Stopping the Avahi daemon..."
+		avahi-daemon -k
+		evaluate_retval
+		;;
+
+	reload)
+		boot_mesg "Reloading the Avahi mDNS/DNS-SD Server Conf. Daemon..."
+		reloadproc avahi-dnsconfd -r
+
+		boot_mesg "Reloading the Avahi daemon..."
+		reloadproc avahi-daemon -r
+		;;
+	restart)
+		${0} stop
+		sleep 1
+		${0} start
+		;;
+
+	status)
+		statusproc avahi-daemon
+		statusproc avahi-dnsconfd
+		;;
+	*)
+		echo "Usage: ${0} {start|stop|reload|restart|status}"
+		exit 1
+		;;
+esac
+
+# End $rc_base/init.d/avahi
diff -Naur avahi-0.6.25.orig/initscript/lfs/avahi.in avahi-0.6.25/initscript/lfs/avahi.in
--- avahi-0.6.25.orig/initscript/lfs/avahi.in	2010-04-03 11:31:54.872916790 -0500
+++ avahi-0.6.25/initscript/lfs/avahi.in	1969-12-31 18:00:00.000000000 -0600
@@ -1,49 +0,0 @@
-#!/bin/sh
-########################################################################
-# Begin $rc_base/init.d/avahi
-#
-# Description : Avahi daemon loader
-#
-# Authors     : William Immendorf - will.immendorf@gmail.com
-#
-# Version     : 00.00
-#
-# Notes       : Based off of the LFS 6.4 sysklogd script.
-#
-########################################################################
-
-. /etc/sysconfig/rc
-. ${rc_functions}
-
-case "${1}" in
-	start)
-		boot_mesg "Starting the Avahi daemon..."
-		loadproc avahi-daemon -D
-		;;
-
-	stop)
-		boot_mesg "Stopping the Avahi daemon..."
-		avahi-daemon -k
-		evaluate_retval
-		;;
-
-	reload)
-		boot_mesg "Reloading the Avahi daemon..."
-		reloadproc avahi-daemon -r
-		;;
-	restart)
-		${0} stop
-		sleep 1
-		${0} start
-		;;
-
-	status)
-		statusproc avahi-daemon
-		;;
-	*)
-		echo "Usage: ${0} {start|stop|reload|restart|status}"
-		exit 1
-		;;
-esac
-
-# End $rc_base/init.d/avahi
diff -Naur avahi-0.6.25.orig/initscript/lfs/Makefile.am avahi-0.6.25/initscript/lfs/Makefile.am
--- avahi-0.6.25.orig/initscript/lfs/Makefile.am	2010-04-03 11:31:54.872916790 -0500
+++ avahi-0.6.25/initscript/lfs/Makefile.am	2010-04-03 11:30:58.904916875 -0500
@@ -19,11 +19,18 @@
 
 initddir = $(sysconfdir)/rc.d/init.d
 
-EXTRA_DIST = avahi.in
-
 initd_SCRIPTS = avahi
 
 CLEANFILES = avahi
 
-avahi: avahi.in
-	chmod +x $@
+avahi:
+	chmod +x avahi
+
+install: install-am
+	ln -sf ../init.d/avahi /etc/rc.d/rc0.d/K28avahi
+	ln -sf ../init.d/avahi /etc/rc.d/rc1.d/K28avahi
+	ln -sf ../init.d/avahi /etc/rc.d/rc2.d/K49avahi
+	ln -sf ../init.d/avahi /etc/rc.d/rc3.d/S21avahi
+	ln -sf ../init.d/avahi /etc/rc.d/rc4.d/S21avahi
+	ln -sf ../init.d/avahi /etc/rc.d/rc5.d/S21avahi
+	ln -sf ../init.d/avahi /etc/rc.d/rc6.d/K28avahi
